### PR TITLE
[Bugfix] Fix a bug that was causing `clustermgtd` to fail when a field returned by the command `scontrol show nodes` has a value that contains the character `=`.

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
Fix a bug that was causing `clustermgtd` to fail when a field returned by the command `scontrol show nodes` has a value that contains the character `=`.

In particular, before this change when the field `Reason` is set to a string containing `=`, then clustermgtd would fail with the below error casuing the nodes management loop to be interrupted.

```
Unable to get partition/node info from slurm, no other action can be performed. Sleeping... Exception: too many values to unpack (expected 2)
```

### Tests
* Unit Test

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/common/schedulers/test_slurm_commands.py [hunk 1]
```diff
@@ -194,23 +194,23 @@ def test_is_static_node(nodename, expected_is_static):
             False,
         ),
         (
-            "NodeName=multiple-dy-c5xlarge-4\n"
-            "NodeAddr=multiple-dy-c5xlarge-4\n"
-            "NodeHostName=multiple-dy-c5xlarge-4\n"
+            "NodeName=multiple-dy-c5xlarge-3\n"
```
Comment (hgreebe) [inline tests/common/schedulers/test_slurm_commands.py:197]: Why did you change the name from `multiple-dy-c5xlarge-4` to `multiple-dy-c5xlarge-3`?

**Review by hgreebe — APPROVED** [review]

File: tests/common/schedulers/test_slurm_commands.py (hunk 1 above)
Comment (gmarciani) [reply tests/common/schedulers/test_slurm_commands.py:197]: no reason, just fake unit test data
